packages/web/src/app/Layout.tsx

```diff
@@ -1,14 +1,42 @@
-import { Suspense } from 'react';
-import { Outlet } from 'react-router-dom';
+import { GetLanguagesResponseBody, Language } from '@translation/api-types';
+import { Suspense, useMemo } from 'react';
+import { Outlet, useLoaderData, useOutletContext } from 'react-router-dom';
+import apiClient from '../shared/apiClient';
+import useLocalStorageState from '../shared/useLocalStorageState';
 import Header from './Header';
 
+export function layoutLoader() {
+  return apiClient.languages.findAll();
+}
+
+export interface LayoutContext {
+  language?: Language;
+}
+
+export function useLayoutContext() {
+  return useOutletContext<LayoutContext>();
+}
+
 export function Layout() {
+  const languages = useLoaderData() as GetLanguagesResponseBody;
+  const [language, selectLanguage] = useLocalStorageState<string>(
+    'translation-language',
+    languages.data[0]?.id
+  );
```

Comment:
We can then use that hook to manage the selected language state in the root route component.